File: Android/TRTCScenesDemo/trtcliveroom/src/main/java/com/tencent/liteav/liveroom/model/TRTCLiveRoomDelegate.java
package com.tencent.liteav.liveroom.model;

import java.util.List;

public interface TRTCLiveRoomDelegate {

    /**
     * 组件出错信息，请务必监听并处理
     * @param code
     * @param message
     */
    void onError(int code, String message);

    /**
     * 组件告警信息
     * @param code
     * @param message
     */
    void onWarning(int code, String message);

    /**
     * 组件log信息
     * @param message
     */
    void onDebugLog(String message);

    /**
     * 房间信息改变的通知
     * @param roomInfo 房间信息
     */
    void onRoomInfoChange(TRTCLiveRoomDef.TRTCLiveRoomInfo roomInfo);

    /**
     * 房间被销毁，当主播
     * @param roomId
     */
    void onRoomDestroy(String roomId);

    /**
     * 主播进入房间, 你可以调用 startPlay() 启动播放用户视频
     * @param userId
     */
    void onAnchorEnter(String userId);

    /**
     * 主播离开房间, 你可以调用 stopPlay() 停止播放用户视频
     * @param userId
     */
    void onAnchorExit(String userId);

    /**
     * 观众进入房间
     */
    void onAudienceEnter(TRTCLiveRoomDef.TRTCLiveUserInfo userInfo);

    /**
     * 观众离开房间
     */
    void onAudienceExit(TRTCLiveRoomDef.TRTCLiveUserInfo userInfo);

    /**
     * 主播收到连麦请求
     * @param userInfo 连麦对象
     * @param reason 原因
     * @param timeOut
     */
    void onRequestJoinAnchor(TRTCLiveRoomDef.TRTCLiveUserInfo userInfo, String reason, int timeOut);

    /**
     * 观众收到踢出连麦
     */
    void onKickoutJoinAnchor();

    /**
     * 主播收到PK请求
     * @param userInfo
     * @param timeout
     */
    void onRequestRoomPK(TRTCLiveRoomDef.TRTCLiveUserInfo userInfo, int timeout);

    /**
     * 退出PK
     */
    void onQuitRoomPK();

    /**
     * 收到文字消息
     * @param roomId
     * @param message
     * @param userInfo
     */
    void onRecvRoomTextMsg(String roomId, String message, TRTCLiveRoomDef.TRTCLiveUserInfo userInfo);

    /**
     * 收到自定义消息
     * @param roomId
     * @param cmd
     * @param message
     * @param userInfo
     */
    void onRecvRoomCustomMsg(String roomId, String cmd, String message, TRTCLiveRoomDef.TRTCLiveUserInfo userInfo);

}

File: iOS/TRTCScenesDemo/TRTCScenesDemo/TRTCLiveRoomDemo/model/TRTCLiveRoomDelegate.swift
//
//  TRTCLiveRoomDelegate.swift
//  trtcScenesDemo
//
//  Created by Xiaoya Liu on 2020/2/8.
//  Copyright © 2020 xcoderliu. All rights reserved.
//

import UIKit

@objc public protocol TRTCLiveRoomDelegate: class {
    /// 出错回调
    @objc optional func trtcLiveRoom(_ trtcLiveRoom: TRTCLiveRoomImpl, onError code: Int, message: String?)
    
    /// 警告回调
    @objc optional func trtcLiveRoom(_ trtcLiveRoom: TRTCLiveRoomImpl, onWarning code: Int, message: String?)
    
    /// 日志回调
    @objc optional func trtcLiveRoom(_ trtcLiveRoom: TRTCLiveRoomImpl, onDebugLog log: String)

    /// 房间销毁回调
    @objc optional func trtcLiveRoom(_ trtcLiveRoom: TRTCLiveRoomImpl, onRoomDestroy roomID: String)

    /// 直播房间信息变更回调
    @objc optional func trtcLiveRoom(_ trtcLiveRoom: TRTCLiveRoomImpl, onRoomInfoChange info: TRTCLiveRoomInfo)

    /// 主播进房回调
    /// - Note: 主播包括房间大主播、连麦观众和跨房PK主播
    @objc optional func trtcLiveRoom(_ trtcLiveRoom: TRTCLiveRoomImpl, onAnchorEnter userID: String)
    
    /// 主播离开回调
    /// - Note: 主播包括房间大主播、连麦观众和跨房PK主播
    @objc optional func trtcLiveRoom(_ trtcLiveRoom: TRTCLiveRoomImpl, onAnchorExit userID: String)

    /// 观众进房回调
    @objc optional func trtcLiveRoom(_ trtcLiveRoom: TRTCLiveRoomImpl, onAudienceEnter user: TRTCLiveUserInfo)

    /// 观众离开回调
    @objc optional func trtcLiveRoom(_ trtcLiveRoom: TRTCLiveRoomImpl, onAudienceExit user: TRTCLiveUserInfo)

    /// 主播收到观众的连麦申请
    @objc optional func trtcLiveRoom(_ trtcLiveRoom: TRTCLiveRoomImpl, onRequestJoinAnchor user: TRTCLiveUserInfo, reason: String?, timeout: Double)

    /// 观众收到主播发来的下麦通知
    @objc optional func trtcLiveRoomOnKickoutJoinAnchor(_ trtcLiveRoom: TRTCLiveRoomImpl)
    
    /// 主播收到其他主播的跨房PK申请
    @objc optional func trtcLiveRoom(_ trtcLiveRoom: TRTCLiveRoomImpl, onRequestRoomPK user: TRTCLiveUserInfo, timeout: Double)

    /// 主播收到PK中对方主播结束PK的通知
    @objc optional func trtcLiveRoomOnQuitRoomPK(_ trtcLiveRoom: TRTCLiveRoomImpl)

    /// 房间成员收到群发的文本消息
    @objc optional func trtcLiveRoom(_ trtcLiveRoom: TRTCLiveRoomImpl, onRecvRoomTextMsg message: String, fromUser user: TRTCLiveUserInfo)
    
    /// 房间成员收到群发的自定义消息
    @objc optional func trtcLiveRoom(_ trtcLiveRoom: TRTCLiveRoomImpl, onRecvRoomCustomMsg command: String, message: String, fromUser user: TRTCLiveUserInfo)
}
